pkgs/stdenv/generic/setup.sh

```diff
@@ -660,15 +660,10 @@ export NIX_INDENT_MAKE=1
 # means that we're supposed to try and auto-detect the number of
 # available CPU cores at run-time.
 
-if [ -z "${NIX_BUILD_CORES:-}" ]; then
-  NIX_BUILD_CORES="1"
-elif (( NIX_BUILD_CORES <= 0 )); then
-  NIX_BUILD_CORES=$(nproc 2>/dev/null || true)
-  if expr >/dev/null 2>&1 "$NIX_BUILD_CORES" : "^[0-9][0-9]*$"; then
-    :
-  else
-    NIX_BUILD_CORES="1"
-  fi
+NIX_BUILD_CORES="${NIX_BUILD_CORES:-1}"
+if ((NIX_BUILD_CORES <= 0 )); then
+  guess=$(nproc 2>/dev/null || true)
+  (( NIX_BUILD_CORES = guess <= 0 ? 1 : guess))
```

Comment:
I’m not usually one to complain about whitespace style, but that’s the most inconsistent possible arrangement of spaces inside the `((…))` statements. ☺

```suggestion
if ((NIX_BUILD_CORES <= 0)); then
  guess=$(nproc 2>/dev/null || true)
  ((NIX_BUILD_CORES = guess <= 0 ? 1 : guess))
```

Reply:
haha, happy to fix this. I'll squash the commits after.